azure-pipelines-templates/job/deploy-api.yml

```diff
@@ -0,0 +1,49 @@
+parameters:
+  DeploymentName: ''
```

Comment:
template parameters default values should be left blank unless the template is designed to succeed with an empty string as the value for a particular parameter

Reply:
So `DeploymentCondition: false` is good but the other parameters appear to be required params rather than optional so should be `DeploymentName: ` rather than `DeploymentName: ''`

Reply:
oooo, I'll get them changed in the platform-building blocks as well then as they also have an empty string as the default when it shouldnt